src/main/java/org/elasticsearch/index/engine/EngineConfig.java

```diff
@@ -167,6 +176,41 @@ public EngineConfig(ShardId shardId, boolean optimizeAutoGenerateId, ThreadPool
         failEngineOnCorruption = indexSettings.getAsBoolean(INDEX_FAIL_ON_CORRUPTION_SETTING, true);
         failOnMergeFailure = indexSettings.getAsBoolean(INDEX_FAIL_ON_MERGE_FAILURE_SETTING, true);
         gcDeletesInMillis = indexSettings.getAsTime(INDEX_GC_DELETES_SETTING, EngineConfig.DEFAULT_GC_DELETES).millis();
+        versionMapSizeSetting = indexSettings.get(INDEX_VERSION_MAP_SIZE, DEFAULT_VERSION_MAP_SIZE);
+        updateVersionMapSize();
+    }
+
+    /** updates {@link #versionMapSize} based on current setting and {@link #indexingBufferSize} */
+    private void updateVersionMapSize() {
```

Comment:
Because we don't trigger a refresh here, if the versionMapSize decreased to a value lower than amount of RAM versionMap is now using, we won't free up that RAM until the next indexing op comes to this engine?  Not sure it matters (will users expect to use this to "free up RAM used by indexer"?).